crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -721,17 +729,6 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                 }
                 Ok(forked_state)
             })
-            // Drop any forked states whose markers are known to never
-            // match any marker environments.
-            .filter(|result| {
-                if let Ok(ref forked_state) = result {
-                    let markers = forked_state.env.try_markers().expect("is a fork");
-                    if markers.is_false() {
-                        return false;
-                    }
-                }
-                true
-            })
```

Comment:
Where did this one go?

Reply:
No longer needed! Since we should no longer generate these forks in the first place.

The key change (below) is from this:

```rust
       let mut forks = vec![Fork {
            dependencies: vec![],
            markers: MarkerTree::TRUE,
        }];
```

to this:

```rust
        let mut forks = vec![Fork {
            dependencies: vec![],
            env: env.clone(),
        }];
```

where `env` contains the markers from the _parent fork_ instead of just using `MarkerTree::TRUE` and then trying to "combine" the markers with the parent fork at a higher level. This awkward dance is ultimately what led to https://github.com/astral-sh/uv/issues/8676. This commit actually undoes [my fix](https://github.com/astral-sh/uv/pull/8759) and instead fixes it by simply not generating the forks in the first place.

I don't know why we didn't always just start with the parent's markers when creating forks. My very vague recollection is that, at the time, we didn't have a good marker implementation and it led to bugs because it would overall produce more "complicated" markers that our disjoint checking failed on.